src/mmw/js/src/data_catalog/models.js

```diff
@@ -29,6 +29,23 @@ var Catalog = Backbone.Model.extend({
         error: '',
     },
 
+    searchIfNeeded: function(query, fromDate, toDate, bounds) {
+        var self = this,
+            isSameSearch = query === this.get('query') &&
+                           fromDate === this.get('fromDate') &&
+                           toDate === this.get('toDate') &&
+                           utils.formatBounds(bounds) === this.get('bbox');
+
+        if (!isSameSearch) {
```

Comment:
Consider the case when a search is already running. If a user were to change the query and re-search, `!isSameSearch` evaluates to `true` and we return `this.searchPromise`. However, when the first search returns, the `always` block executes and deletes `self.searchPromise`. What happens if this were to be deleted before the client can use its value?